xbmc/threads/Thread.cpp

```diff
@@ -102,6 +102,7 @@ void CThread::Create(bool bAutoDelete, unsigned stacksize)
 
 bool CThread::IsRunning() const
 {
+  CSingleLock lock(m_CriticalSection);
```

Comment:
No, it is not undefined behaviour because the value of m_ThreadId is not of any interest. Lets assume it is only half written and as a result not null. The result of IsRunning would be true. A few processor cycles later the result would be false. As already mentioned several times here: The result of InRunning might not be valid anymore after the call. Neither a lock nor an atomic would change this fact.

Please construct a scenario where a lock would change behaviour. I bet you can't.